packages/@ember/-internals/routing/lib/services/router.ts

```diff
@@ -62,6 +62,12 @@ function cleanURL(url: string, rootURL: string) {
  */
 export default class RouterService extends Service {
   _router!: EmberRouter;
+  get router() {
```

Comment:
I worry about directly exposing the private router here. If folks do `this.router.router` now they have access to the `router:main` thing, which doesn't seem like exactly what we want 🤔.

I do like forcing all of our callers to go through the getter though 🤔

Reply:
Maybe `get privateRouter()`? Folks do have access to `_router` anyways, it's mostly about naming it to callout the private access.

Reply:
>  Folks do have access to _router anyways, it's mostly about naming it to callout the private access.

Ya, exactly. But IMHO, at that point why do we need to keep two properties? I'm wondering if we can remove the auto-injection of `_router` (as `router:main`) from here:

https://github.com/emberjs/ember.js/blob/3ef84958940e5aa7e551eb0404b3966c8c9728b2/packages/%40ember/application/lib/application.js#L1170

Instead, we could do this:

```js
const ROUTER = symbol('ROUTER'); // uses the internal symbol utility method

// ...snip...

get _router() {
  const router = this[ROUTER];
  if (router !== undefined) {
    return router;
  }

  let owner = getOwner(this);
  owner.setupRouter();

  return this[ROUTER] = owner.router;
}
```

Reply:
Change a bit to do `owner.lookup('router:main')`, because in `RenderingTestCase` it appears there's no `owner.router`.
`router:main` is also injected into `service:-routing`, but I'm skipping those since I'm not sure if it's safe to change engine.